magento/pwa-studio
# When auth token is invalid reset cart and refetch details

## Description

The retry logic was not functioning when a guest user tried to perform an operation on an "authed" cart. This situation could arise if a user signs in and then the auth token expires naturally (1 hour ttl). The old, authed, cart id persisted and was used but since there was no more token in store the request failed.

## Related Issue





Closes #2378 .

## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps


1. Log in and add an item to your cart.
2. In local storage, change the `M2_VENIA_BROWSER_PERSISTENCE__signin_token` entry to have a `timeStored` of some low number like `1`.
3. Add another item to your cart. The operation should succeed but the cart should only have the item you just added. The cart trigger should be updated to show `1` item in the cart.

**Rerun the above steps but on step 3 _refresh_ the page and then add an item.**

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.
